### PR TITLE
feat(workflows) #26839 : Expose Workflow Information Through Rest API

### LINKED ISSUE #26839: Edit Content: Expose Workflow Information Through Rest APIs
### Parent
#25445 

### User Story
As a dotCMS user, I want to be able to view the workflow information and step a piece of content is currently in, so I can better understand the content lifecycle.

```[tasklist]
### Acceptance Criteria
- [x] Create a REST endpoint to return the workflow information for a piece of content.
- [x] Ensure the endpoint is secured and only accessible to authenticated users.
- [x] Add documentation on how the API works in the ticket so it can be documentaed by @jdcmsd
```

### Proposed Objective
Core Feature

### Proposed Priority
Priority 3 - Average

### External Links... Slack Conversations, Support Tickets, Figma Designs, etc.
N/A

### Assumptions & Initiation Needs
- The API endpoint should be able to handle requests from any authenticated user.

### Quality Assurance Notes & Workarounds
N/A

### Sub-Tasks & Estimates
- Create a REST endpoint to return the workflow information for a piece of content: 5 hours 
- Ensure the endpoint is secured and only accessible to authenticated users: 2 hours

### PR DESCRIPTION
**Pull request by jcastro-dotcms**
### Proposed Changes
* Exposes a new service that allows you to get information on the current status of a Contentlet in its current Workflow Step. This is specially useful for the Angular UI to display the Contentlet status that we currently see on the right pane when editing a piece of content.

### <samp>🤖[[deprecated]](https://githubnext.com/copilot-for-prs-sunset) Generated by Copilot at 53fe249</samp>

This pull request refactors several classes related to the dotCMS Workflow system, such as `WorkflowResource`, `WorkflowScheme`, `WorkflowTask`, and `WorkflowStep`, to improve their readability and maintainability. It also adds a new REST endpoint to `WorkflowResource` that returns a `ContentletWorkflowStatusView` object, which provides a simplified view of the status of a Contentlet in a Workflow. This endpoint is used by the back-end UI to display the Workflow information for a Contentlet.

### CODE REVIEW

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/workflow/ContentletWorkflowStatusView.java [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+import com.dotmarketing.util.UtilMethods;
+
+/**
+ * Provides a simplified view of the status of a Contentlet in a dotCMS Workflow. This is used by
+ * the back-end UI, and can be extended as rquired.
+ *
+ * @author Jose Castro
+ * @since Dec 4th, 2023
+ */
+public class ContentletWorkflowStatusView {
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotcms/rest/api/v1/workflow/ContentletWorkflowStatusView.java:19]: make serializable

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/workflow/WorkflowResource.java [hunk 2]
```diff
@@ -3166,6 +3170,68 @@ public final void deleteScheme(@Context final HttpServletRequest request,
...
+                "'%s'", contentletInode));
+        final InitDataObject initDataObject = this.webResource.init
+                (null, request, response, true, null);
+        final User user = initDataObject.getUser();
+        WorkflowStep wfStep = null;
+        WorkflowScheme scheme = null;
+        WorkflowTask wfTask = new WorkflowTask();
+        try {
+            final Contentlet contentlet = this.contentletAPI.find(contentletInode, user, false);
+            final List<WorkflowStep> stepsByContentlet =
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotcms/rest/api/v1/workflow/WorkflowResource.java:3210]: isnt this one WorkflowAPI.findCurrentStep ???

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/workflow/ContentletWorkflowStatusView.java (hunk 1 above)
Comment (jcastro-dotcms) [reply dotCMS/src/main/java/com/dotcms/rest/api/v1/workflow/ContentletWorkflowStatusView.java]: Done!

**Review by jdotcms — APPROVED** [review]

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/workflow/WorkflowResource.java [hunk 3]
```diff
@@ -3166,6 +3170,68 @@ public final void deleteScheme(@Context final HttpServletRequest request,
...
+     * @param contentletInode The inode of the Contentlet whose status will be checked.
+     *
+     * @return The status information of the Contentlet in the Workflow it is assigned to.
+     */
+    @GET
+    @Path("/status/{contentletInode}")
+    @JSONP
+    @NoCache
+    @Produces({MediaType.APPLICATION_JSON, "application/javascript"})
+    public final Response getStatusForContentlet(@Context final HttpServletRequest request,
```
[the remark this answers is not in the document]
Comment (jcastro-dotcms) [reply dotCMS/src/main/java/com/dotcms/rest/api/v1/workflow/WorkflowResource.java]: Done!

**Review by erickgonzalez — APPROVED** [review]

**Pull request merged**